### PR TITLE
Add "grid" as a tag to the output file template for CMIP6

### LINKED ISSUE #1134: CMIP6 datasets that only differ in grid cannot be used in a single recipe variable section
As reported by @ledm in https://github.com/ESMValGroup/ESMValTool/issues/2131, CMIP6 datasets that only differ in grid cannot be used in the same variable section in a recipe. Is this actually a feature you would want like to have @ledm?

This is because the grid is not part of the output file DRS:
https://github.com/ESMValGroup/ESMValCore/blob/347d19d28c0b91891180a58a211ad7781c2d6999/esmvalcore/config-developer.yml#L29
so the files would overwrite each other and therefore it's not allowed. The solution would be to add the grid to the output file name.

### PR DESCRIPTION
**Pull request by zklaus**
## Description

Closes #1134 

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [ ] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [ ] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/config-developer.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ CMIP6:
     ETHZ: '{exp}/{mip}/{short_name}/{dataset}/{ensemble}/{grid}/'
     SYNDA: '{activity}/{institute}/{dataset}/{exp}/{ensemble}/{mip}/{short_name}/{grid}/{latestversion}'
   input_file: '{short_name}_{mip}_{dataset}_{exp}_{ensemble}_{grid}*.nc'
-  output_file: '{project}_{dataset}_{mip}_{exp}_{ensemble}_{short_name}'
+  output_file: '{project}_{dataset}_{mip}_{exp}_{ensemble}_{short_name}_{grid}'
```
Comment (valeriupredoi) [inline esmvalcore/config-developer.yml:31]: I think this needs a bit more attention: is `grid` now a mandatory parameter? If it is, we need to add an exception in `_recipe.py` in the form of a `RecipeError` if it's not provided

Comment (zklaus) [reply esmvalcore/config-developer.yml:31]: I think for CMIP6 grid has always been mandatory. Do you have a recipe that runs without a specified grid?

Comment (valeriupredoi) [reply esmvalcore/config-developer.yml:31]: we can do that later, in a new PR if you think the current behaviour is enough for a new though :+1:

Comment (bouweandela) [reply esmvalcore/config-developer.yml:31]: You're right we could catch this error before even looking for data, could you open a new issue suggesting that as an improvement?

Comment (valeriupredoi) [reply esmvalcore/config-developer.yml:31]: > I changed that recently, look at the code above. You may need to do a git pull if you still get a KeyError.

ah awesome, cheers!

> You're right we could catch this error before

Shall dos! But isn't that a trivial fix by adding `grid` to the `required_keys` list in `_recipe.py` (if we want to promote it to being so important, that is)

Comment (zklaus) [reply esmvalcore/config-developer.yml:31]: Feel free to put that information in the issue.

Comment (bouweandela) [reply esmvalcore/config-developer.yml:31]: Yes, but we could also be a bit smarter and extract `required_keys` from the project DRS

**Pull request merged**